### PR TITLE
VNF: add API listVnfAppliances

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes: #8979 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java [hunk 1]
```diff
@@ -1409,6 +1410,12 @@ private Pair<List<Long>, Integer> searchForUserVMIdsAndCount(ListVMsCmd cmd) {
         }
 
         Boolean isVnf = cmd.getVnf();
+        if (cmd instanceof ListVnfAppliancesCmd) {
+            if (isVnf != null && !isVnf) {
+                throw new InvalidParameterValueException("Unable to list VNF appliances as isvnf is set to false");
+            }
+            isVnf = true;
+        }
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:1418]: shouldn't the command class set the falg to true in it's constructor and not allow it to be changed?

Comment (weizhouapache) [reply server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:1418]: if `isVnf` is set to false, should throw an exception ?

Comment (DaanHoogland) [reply server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:1418]: well, if we get the listVnfAppliences API call we shouldn't bother about the flag at all, I think. The API implies the flag is true.

Comment (weizhouapache) [reply server/src/main/java/com/cloud/api/query/QueryManagerImpl.java]: pr is updated. thanks @DaanHoogland

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test rocky8 kvm-rocky8

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by weizhouapache** [comment]
@vladimirpetrov 
can you verify when possible ? thanks

**Pull request merged**